Implement managed thumbnail request #3227

ashklianko opened the pull request:
-Sending XHR request to control image displaying in grid
-Caching image request for specified image id
-Showing image placeholder until 200 server response received
-Converting image received to a base64 string and setting to an Img tag src to prevent second image request (img src tag doesn't use previously received xhr response)
-In future failed request to be managed specifically

alansemenov committed:
Added WidgetResource (#2947)

alansemenov committed:
Removed StatusResource

rymsha committed:
copied resources (all, not just widget)
fixed url to app widget icons

alansemenov committed:
Moved `commons-text` dependency to the root project

(cherry picked from commit 66fa394bb268f6e841b03250cbb4926e32668b4a)

alansemenov committed:
Moved `commons-text` dependency to the lib `module`

(cherry picked from commit ede0cae2c7a921f8035233f3df74ae412b560541)

rymsha committed:
Don't use app plugin for libs #2952

(cherry picked from commit 1cc8fba7ddf9d7b9b6920e1c5e45d9f19da42664)

sgauruseu committed:
ui-tests issue#3148

ashklianko committed:
Page Component View - incorrect component name for the duplicated text component #3160

ashklianko committed:
IsAuthenticatedRequest: cache LoginResult? #3224

-Removing saved LoginResult values because now classes will be using auth request cache

ashklianko committed:
Responsive Widget panel in docked mode #3226

-Docked panel put inside SplitPanel with live/statistics panel to occupy 68% of available width
-LivePanel related code moved to ContentWizardPanel

alansemenov committed:
Update ContextSplitPanel.ts

alansemenov committed:
Merge remote-tracking branch 'origin/master' into epic-admin-api

# Conflicts:
#	build.gradle
#	modules/app/build.gradle
#	modules/lib/build.gradle
#	modules/rest/src/test/java/com/enonic/xp/app/contentstudio/rest/resource/status/StatusResourceTest.java

ashklianko committed:
-more fixes

alansemenov committed:
Codacy (#3226)

alansemenov committed:
Codacy (#3226)

alansemenov committed:
Moved ProjectResource and related classes (#3239)

alansemenov committed:
Removed test for StatusResouce (#3239)

ashklianko committed:
Implement managed thumbnail request #3227

-Sending XHR request to control image displaying in grid
-Caching image request for specified image id
-Showing image placeholder until 200 server response received
-Converting image received to a base64 string and setting to an Img tag src to prevent second image request (img src tag doesn't use previously received xhr response)
-In future failed request to be managed specifically

ashklianko committed:
Implement managed thumbnail request #3227

-Sending XHR request to control image displaying in grid
-Caching image request for specified image id
-Showing image placeholder until 200 server response received
-Converting image received to a base64 string and setting to an Img tag src to prevent second image request (img src tag doesn't use previously received xhr response)
-In future failed request to be managed specifically

dependabot[bot] committed:
Bump mockito-core from 3.11.1 to 3.11.2

Bumps [mockito-core](https://github.com/mockito/mockito) from 3.11.1 to 3.11.2.
- [Release notes](https://github.com/mockito/mockito/releases)
- [Commits](https://github.com/mockito/mockito/compare/v3.11.1...v3.11.2)

---
updated-dependencies:
- dependency-name: org.mockito:mockito-core
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ashklianko committed:
Fix for wizard panel init errors

sgauruseu committed:
ui-tests issue#3249

sgauruseu committed:
ui-tests gradle plugin updated

alansemenov committed:
Fixed 500 error when opening Settings via direct link

ashklianko committed:
Content Wizard - display name and path are cleared after opening a content #3248

alansemenov committed:
Project icon fix (#3248)

ashklianko committed:
Insert Panel - missed localization resource for Rich Text Editor #3238

alansemenov committed:
Upgraded Gradle, Node & NPM

rymsha committed:
Fix release automation enonic/xp#8929

alansemenov committed:
Fixed broken layout in the Widget Selector

ashklianko committed:
B/U/I shortcuts in HTML Editor stopped working #3235

-Adding keyboard shortcuts manually for bold, italic, underline styling that will work even wheb B/I/U toolbar buttons missing

sgauruseu committed:
Update failed tests for projects issue#3258

alansemenov committed:
Fixed spacing in the modal dialog's dependant list

ashklianko committed:
B/I/U buttons are missing in HTML Editor's fullscreen mode #3253

-Adding those only if thy are not excluded in config

ashklianko committed:
allowHeadings config not respected when used with exclude/include #3254

-We replaced usage of basic Format plugin with more customizable Styles plugin some time ago, however in docs we have Format plugin mentioned in a list of plugins, thus need to replace Format->Styles when parsing include/exclude items
-Also found an issue: need to filter default toolbar items from the list of include items so they don't appear twice in the toolbar

alansemenov committed:
Updated package-lock.json

alansemenov committed:
Updated package-lock.json

alansemenov committed:
Updated package-lock.json in testing

ashklianko committed:
allowHeadings config not respected when used with exclude/include #3254

-code adjustment

alansemenov committed:
Fixed styling of toggled buttons in Html Editor toolbar (#3253)

ashklianko committed:
-Implementing undo for keyboard shortcuts, so after applying same styling via shortcut style will be removed

sgauruseu committed:
ui-tests issue#3269

ashklianko committed:
Incorrect layout in some of the selectors' dropdown #3273

-Defining combobox grid item viewers height directly where it is needed

alansemenov committed:
Updated docs for <allow-on-content-type> config

ashklianko committed:
Incorrect layout in some of the selectors' dropdown #3273

ashklianko committed:
Content Wizard - Show Context Panel button is not displayed for content without Page Editor #3267

-Adjusted behaviour

jsi committed:
Polish translations by Michał

pmi committed:
Move IssueResource endpoint #3286

pmi committed:
Merge remote-tracking branch 'origin/master' into issue-3286

# Conflicts:
#	modules/app/package-lock.json
#	modules/app/src/main/resources/assets/styles/wizard/content-wizard-toolbar.less
#	modules/lib/package-lock.json
#	modules/lib/src/main/resources/assets/js/app/view/context/NonMobileContextPanelsManager.ts
#	modules/lib/src/main/resources/assets/js/app/view/context/button/NonMobileContextPanelToggleButton.ts
#	testing/page_objects/wizardpanel/details/wizard.details.panel.js
#	testing/page_objects/wizardpanel/liveform/live.form.panel.js
#	testing/specs/page-editor/generate.name.for.fragments.spec.js

pmi committed:
Merge pull request #3296 from enonic/issue-3286

Move IssueResource endpoint #3286

pmi committed:
Move ContentMediaResource endpoint #3300

pmi committed:
Merge remote-tracking branch 'origin/master' into issue-3300

pmi committed:
Merge pull request #3304 from enonic/issue-3300

Move ContentMediaResource endpoint #3300

pmi committed:
Move ContentIconResource endpoint #3305

pmi committed:
Merge pull request #3307 from enonic/issue-3305

Move ContentIconResource endpoint #3305

pmi committed:
Move ContentImageResource endpoint #3309

pmi committed:
Merge pull request #3310 from enonic/issue-3309

Move ContentImageResource endpoint #3309

pmi committed:
Move ContentResource endpoint #3312

pmi committed:
Merge pull request #3319 from enonic/issue-3312

Move ContentResource endpoint #3312

pmi committed:
Move PageResource endpoint #3320

pmi committed:
Merge pull request #3322 from enonic/issue-3320

Move PageResource endpoint #3320

pmi committed:
Move PageTemplateResource endpoint #3323

pmi committed:
Merge pull request #3326 from enonic/issue-3323

Move PageTemplateResource endpoint #3323

pmi committed:
Move PageDescriptorResource endpoint #3327

pmi committed:
Merge remote-tracking branch 'origin/master' into issue-3327

pmi committed:
Merge pull request #3328 from enonic/issue-3327

Move PageDescriptorResource endpoint #3327

pmi committed:
Move LayoutDescriptorResource and PartDescriptorResource endpoints #3330

pmi committed:
Merge pull request #3331 from enonic/issue-3330

Move LayoutDescriptorResource and PartDescriptorResource endpoints #3330

pmi committed:
Move FragmentResource endpoint #3333

sgauruseu committed:
ui-tests - gradle task for github actions  issue#3290

sgauruseu committed:
ui-tests - first test added issue#3290

alansemenov reviewed (changes requested):
* Uploading an image directly via New Content dialog now fails with these errors:
 
![image](https://user-images.githubusercontent.com/11612490/126996281-e25f3981-fc9d-41bf-bd03-de80483e90de.png)
 
* We need to add some kind of animation indicating that the image is being loaded. Right now there's not even an empty placeholder which would be replaced by the loaded image. I suggest showing a small spinner for each image that is being loaded, and making sure that the spinner container is of the same width/height as image thumbnail so that display-name/path won't move once the image is loaded.
Note how the the top 3 images (these are hi-res) in the expanded folder take their time to load with no indication whatsoever, then move the name/path block to the right:

![chrome-capture](https://user-images.githubusercontent.com/11612490/126998190-e5f4efbd-7fb1-4f52-a03a-bc00facc534e.gif)

alansemenov commented:
Images I used for testing: https://drive.google.com/drive/folders/1DjvNbviDzPM5NOIr9CrOokFzrmytaU7U?usp=sharing

pmi committed:
Move FilterByContentResource endpoint #3336

pmi committed:
Merge pull request #3337 from enonic/issue-3333

Move FragmentResource endpoint #3333 and Move FilterByContentResource endpoint #3336

pmi committed:
Move ContentTypeResource endpoint #3338

pmi committed:
Merge remote-tracking branch 'origin/master' into issue-3338

# Conflicts:
#	modules/app/package-lock.json
#	modules/lib/package-lock.json

pmi committed:
Merge pull request #3339 from enonic/issue-3338

Move ContentTypeResource endpoint #3338

pmi committed:
Move XDataResource endpoint #3340

pmi committed:
Merge pull request #3341 from enonic/issue-3340

Move XDataResource endpoint #3340

pmi committed:
Move MixinResource endpoint #3343

pmi committed:
Merge remote-tracking branch 'origin/master' into issue-3343

pmi committed:
Move MixinResource endpoint #3343

pmi committed:
Merge pull request #3348 from enonic/issue-3343

Move MixinResource endpoint #3343

ashklianko committed:
Merge remote-tracking branch 'remotes/origin/master' into issue-3227

pmi committed:
Move MacroResource endpoint #3355

edloidas committed:
Fixed missed endpoints

edloidas committed:
Fix build order #3354

Made `copyDevResources` and JS sources build tasks (`typescript` and `webpack`) depend on lib-admin-ui `build` task to prevent them from starting before the `lib-admin-ui-*-dev-resources.jar` is ready.
Moved `lib-admin.gradle` to `allprojects` to make root tasks to depend on lib-admin-ui too and avoid problems with `copyDevResources` in `app`.
Added `env.gradle` and `node.gradle` to `allprojects`, since they contain definitions that  are used in `lib-admin.gradle`.

alansemenov committed:
Updated package-lock.json

edloidas committed:
Fix build order #3354

Moved `lib-admin.gradle` and other files back to subprojects.

edloidas committed:
Fix build order #3354

Removed excess code.

rymsha committed:
ui-tests in GitHub Actions #3290

pmi committed:
Merge pull request #3356 from enonic/issue-3355

Move MacroResource endpoint #3355

edloidas committed:
Fix build order #3354

Removed excess tasks dependencies.

pmi committed:
Remove Admin API #3363

- remove MixinResource

pmi committed:
Merge pull request #3364 from enonic/issue-3363

Remove Admin API #3363

ashklianko committed:
Implement managed thumbnail request #3227

-showing loading spinner while image is loading, however issue is in that slickgrid rerendering it's content on expand/collapse, thus have to adjust elements via document

ashklianko committed:
Implement managed thumbnail request #3227

-showing loading spinner while image is loading, however issue is in that slickgrid rerendering it's content on expand/collapse, thus have to adjust elements via document

ashklianko committed:
Implement managed thumbnail request #3227

-showing loading spinner while image is loading, however issue is in that slickgrid rerendering it's content on expand/collapse, thus have to adjust elements via document

dependabot[bot] committed:
Bump codecov/codecov-action from 1 to 2.0.2

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 1 to 2.0.2.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v1...v2.0.2)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

pmi committed:
Merge remote-tracking branch 'origin/master' into epic-admin-api

rymsha committed:
Snapshot builds on branches fail enonic/xp-distro#178

ashklianko committed:
Wizard toolbar - button 'Show Component View' should not be visible when a controller is not selected #3294

-checking if content is renderable

alansemenov committed:
Replaced `startsWith` with `indexOf` (#3370)

alansemenov committed:
Switched Webpack from file-loader to asset modules. Removed legacy fonts, switched to woff2 as the main font file. Rearranged image files (#3372)

pmi committed:
Merge remote-tracking branch 'origin/master' into epic-admin-api

jsi committed:
Norwegian and Belarussian translations

alansemenov committed:
A different path to the icon files (#3372)

alansemenov committed:
Don't exclude font files from the jar (#3372)

ashklianko committed:
Incorrect display-name/path overflow in the Content Grid #3375

pmi committed:
Update project sync request to new url

pmi committed:
Move LocaleComboBox and GetLocalesRequest to content studio

alansemenov committed:
Codacy

ashklianko committed:
404 error appears after adding a layout to a page #3317

rymsha committed:
ui-tests in GitHub Actions #3290

rymsha committed:
ui-tests in GitHub Actions #3290

ashklianko committed:
Expander icon is not present in just inserted layout component #3350

-Correctly setting node 'expandable' property and deleting node children on reset

ashklianko committed:
-codacy

ashklianko committed:
-codacy

ashklianko committed:
-codacy

ashklianko committed:
-codacy

ashklianko committed:
-codacy

ashklianko committed:
-codacy

ashklianko committed:
-codacy

pmi committed:
Move WidgetDescriptorServiceImpl to admin-impl #3384

- add docs for new config properties in com.enonic.app.contentstudio.cfg

rymsha committed:
ui-tests in GitHub Actions #3290

pmi committed:
Fix failing build

alansemenov committed:
Merge pull request #3378 from enonic/epic-admin-api

Migrate Admin API from XP runtime

alansemenov committed:
Merge remote-tracking branch 'origin/issue-3227' into issue-3227

alansemenov reviewed (approved):
Nice
